financial-services-accelerator/components/org.wso2.financial.services.accelerator.keymanager/src/main/java/org/wso2/financial/services/accelerator/keymanager/utils/IdentityServerUtils.java

```diff
@@ -0,0 +1,289 @@
+import org.wso2.financial.services.accelerator.common.util.FinancialServicesUtils;
+import org.wso2.financial.services.accelerator.common.util.HTTPClientUtils;
+import org.wso2.financial.services.accelerator.keymanager.internal.KeyManagerDataHolder;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Util class for access Identity server APIs through key manager.
+ */
+public class IdentityServerUtils {
+
+    @SuppressFBWarnings("HTTP_PARAMETER_POLLUTION")
+    public static String getAppIdFromClientId(String clientId) throws FinancialServicesException {
+
+        try {
+            String url = getIdentitySeverUrl() + FSKeyManagerConstants.APP_MGMT_API_URL;
+            URIBuilder builder = new URIBuilder(url);
+            URI uri = builder.build();
+            HttpGet httpGet = new HttpGet(uri.toString() + "?filter=clientId+eq+".concat(clientId));
+
+            String userName = getAPIMgtConfig(FSKeyManagerConstants.API_KEY_VALIDATOR_USERNAME);
+            String password = getAPIMgtConfig(FSKeyManagerConstants.API_KEY_VALIDATOR_PASSWORD);
+            httpGet.setHeader(FinancialServicesConstants.AUTH_HEADER,
+                    FinancialServicesUtils.getBasicAuthHeader(userName, password));
+            CloseableHttpResponse response = HTTPClientUtils.getHttpsClient().execute(httpGet);
+            if (response.getStatusLine().getStatusCode() != 200) {
+                throw new FinancialServicesException("Error while getting app id from client id");
+            }
+            InputStream in = response.getEntity().getContent();
+            JSONObject responseObj = new JSONObject(IOUtils.toString(in, String.valueOf(StandardCharsets.UTF_8)));
+            JSONArray appArray = responseObj.getJSONArray("applications");
+            if (appArray.length() > 0) {
+                return appArray.getJSONObject(0).getString("id");
+            }
+            return null;
+        } catch (IOException e) {
+            throw new FinancialServicesException("Error while getting app id from client id", e);
+        } catch (URISyntaxException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static JSONObject getSPApplicationFromClientId(String clientId) throws FinancialServicesException {
+
+        String appId = getAppIdFromClientId(clientId);
+
+        try {
+            String url = getIdentitySeverUrl() + FSKeyManagerConstants.APP_MGMT_API_URL + appId;
+            URIBuilder builder = new URIBuilder(url);
+            HttpGet httpGet = new HttpGet(builder.build());
+
+            String userName = getAPIMgtConfig(FSKeyManagerConstants.API_KEY_VALIDATOR_USERNAME);
+            String password = getAPIMgtConfig(FSKeyManagerConstants.API_KEY_VALIDATOR_PASSWORD);
```

Comment:
use char[] instead String for password param.https://security.docs.wso2.com/en/latest/security-guidelines/secure-engineering-guidelines/secure-coding-guidlines/general-recommendations-for-secure-coding/